doc/interpolation.rst

```diff
@@ -0,0 +1,183 @@
+.. _interp:
+
+Interpolating data
+==================
+
+.. ipython:: python
+   :suppress:
+
+    import numpy as np
+    import pandas as pd
+    import xarray as xr
+    np.random.seed(123456)
+
+xarray offers flexible interpolating routines, which has a similar interface
+to our :ref:`indexing <indexing>`.
+
+.. note::
+
+  ``interp`` requires `scipy` installed.
+
+
+Scalar and 1-dimensional interpolation
+--------------------------------------
+
+Interpolating a :py:class:`~xarray.DataArray` works mostly like labeled
+indexing of a :py:class:`~xarray.DataArray`,
+
+.. ipython:: python
+
+    da = xr.DataArray(np.sin(0.3 * np.arange(12).reshape(4, 3)),
+                      [('time', np.arange(4)),
+                       ('space', [0.1, 0.2, 0.3])])
+    # label lookup
+    da.sel(time=3)
+
+    # interpolation
+    da.interp(time=3.5)
+
+
+Similar to the indexing, :py:meth:`~xarray.DataArray.interp` also accepts an
+array-like, which gives the interpolated result as an array.
+
+.. ipython:: python
+
+    # label lookup
+    da.sel(time=[2, 3])
+
+    # interpolation
+    da.interp(time=[2.5, 3.5])
+
+.. note::
+
+  Currently, our interpolation only works for gridded array.
+  Therefore, as similar to :py:meth:`~xarray.DataArray.sel`,
+  only dimensional coordinates can be used for the original coordinate to be
```

Comment:
In theory it should work with non-dimension coordinates with the same shape and dimension as the dimension coordinate, right? (http://xarray.pydata.org/en/latest/data-structures.html#coordinates)

Reply:
Yes, it should work in theory. But currently, it is not supported, just because of the analogy to our `.sel()`.
To support this, we may need to make an additional design decision, 
+ which coordinate should be the dimensional coordinate if the array is interpolated along non-dimensional coordinate?
+ how should we do if the argument is a DataArray?

I suppose in the future we may support unstructured data, where we interpolate data along non-dimensional coordinates.
I think it is probably cleaner that the current `.interp()` only supports dimensional coordinates, and leave the interpolation along non-dimensional coordinates to the future implementation of the unstructured array support.